src/vm/ilmarshalers.h

```diff
@@ -618,6 +618,41 @@ class ILMarshaler
 #else
                 byrefNativeReturn = true;
 #endif
+                // We need to adjust the treatment of a struct with one field to be the type of the field
```

Comment:
Are we going to provide some way for the user to indicate they want this marshaled correctly?

Is this only violated for the COM case or also for the explicit `CallingConvention.ThisCall` case?

Reply:
It’s for both COM and ThisCall. There’s a pretty easy way to disable this workaround. Make your strict explicit and make a second field of the same type as the first field at the same offset. It’ll match the layout of the original type and opt-out of this workaround.

The test updates in this PR have 2 examples of this workaround.

Reply:
>Are we going to provide some way for the user to indicate they want this marshaled correctly?

At this point probably not.

>Is this only violated for the COM case or also for the explicit `CallingConvention.ThisCall` case?

This was discovered in a COM scenario, but would impact the `CallingConvention.ThisCall` case as well.

Reply:
> Make your strict explicit and make a second field of the same type as the first field at the same offset.

Isnt this more expensive for a number of cases? IIRC, we cant optimize this as well and it will cause both fields to be copied/handled in a number of cases.

Reply:
It’s only more expensive (at least in the interop case) if your structure has one field that is (after unwrapping the single-field structs) a non-blittable primitive. So only structs that have a single field of type bool or an ANSI char.